src/Writer/PdoWriter.php

```diff
@@ -52,33 +67,44 @@ public function __construct(\PDO $pdo, $tableName)
      */
     public function writeItem(array $item)
     {
-        try {
-            //prepare the statment as soon as we know how many values there are
-            if (!$this->statement) {
-
+        if (null === $this->statement) {
+            try {
                 $this->statement = $this->pdo->prepare(sprintf(
-                    'INSERT INTO %s(%s) VALUES (%s)',
+                    'INSERT INTO %s (%s) VALUES (%s)',
                     $this->tableName,
                     implode(',', array_keys($item)),
                     substr(str_repeat('?,', count($item)), 0, -1)
                 ));
-
-                //for PDO objects that do not have exceptions enabled
-                if (!$this->statement) {
-                    throw new WriterException('Failed to prepare write statement for item: '.implode(',', $item));
-                }
+            } catch (\PDOException $e) {
+                throw new WriterException('Failed to send query', null, $e);
             }
+        }
 
-            //do the insert
-            if (!$this->statement->execute(array_values($item))) {
-                throw new WriterException('Failed to write item: '.implode(',', $item));
-            }
+        $this->stack[] = array_values($item);
 
-        } catch (\Exception $e) {
-            //convert exception so the abstracton doesn't leak
-            throw new WriterException(sprintf('Write failed (%s)', $e->getMessage()), null, $e);
-        }
+        return $this;
+    }
+
+    public function finish()
+    {
+        $this->flush();
```

Comment:
This is the number one risk of adding a flush method: `flush` is called twice if used in a batch writer: once when flush is called by the batch, once when called in finish.